app/src/LibraryUtils.cpp

```diff
@@ -0,0 +1,433 @@
+			        NULL);
+			if (allowedValueRefResult->getNodeValue() == NULL)
+			{
+				LOG_ERROR() << "There is no result for the provided XPath " << std::endl;
+			}
+			else
+			{
+				XMLSize_t allowedValueCount = allowedValueRefResult->getSnapshotLength();
+				for (XMLSize_t allowedValueSize = 0; allowedValueSize < allowedValueCount; allowedValueSize++)
+				{
+					allowedValueRefResult->snapshotItem(allowedValueSize);
+					DOMNode* allowedValueItem = allowedValueRefResult->getNodeValue();
+					//The allowed values has to be updated.
+					/*Result res=OpenConfiguratorCore::GetInstance().SetParameterAllowedValues(ValidateCLIParameter::GetInstance().GetNetworkName(),nodeId,uniqueID,vector<> allowedValues);
+
+					if(!res.IsSuccessful()) {
+						LOG_ERROR() << res.GetErrorMessage();
+					}*/
+				}
+			}
+
+//defaultvalue update
+
+			DOMXPathResult* defaultValueRefResult = doc->evaluate(
+			        XMLString::transcode("./defaultValue"),
+			        parameterTemplateItem,
+			        NULL,
+			        DOMXPathResult::ORDERED_NODE_SNAPSHOT_TYPE,
+			        NULL);
+			if (defaultValueRefResult->getNodeValue() == NULL)
+			{
+				std::cout << "There is no result for the provided XPath " << std::endl;
+			}
+			else
+			{
+				DOMNode* defaultValueItem = defaultValueRefResult->getNodeValue();
+				char* defaultValue = XMLString::transcode(defaultValueItem->getAttributes()->getNamedItem(XMLString::transcode("value"))->getNodeValue());
+				std::string defaultVal = (std::string)defaultValue;
+
+				/*Result res=OpenConfiguratorCore::GetInstance().SetParameterDefaultValue(ValidateCLIParameter::GetInstance().GetNetworkName(),nodeId,uniqueID,defaultVal);
+
+				if(!res.IsSuccessful()) {
+					LOG_ERROR() << res.GetErrorMessage();
+				}*/
+
+			}
+
+
+
+			DOMXPathResult* actualValueRefResult = doc->evaluate(
+			        XMLString::transcode("./actualValue"),
+			        parameterTemplateItem,
+			        NULL,
+			        DOMXPathResult::ORDERED_NODE_SNAPSHOT_TYPE,
+			        NULL);
+			if (actualValueRefResult->getNodeValue() == NULL)
+			{
+				std::cout << "There is no result for the provided XPath " << std::endl;
+			}
+			else
+			{
+				DOMNode* actualValueItem = actualValueRefResult->getNodeValue();
+				char* actualValue = XMLString::transcode(actualValueItem->getAttributes()->getNamedItem(XMLString::transcode("value"))->getNodeValue());
+				std::string actualtVal = (std::string)actualValue;
+
+				/*Result res=OpenConfiguratorCore::GetInstance().SetParameterActualValue(ValidateCLIParameter::GetInstance().GetNetworkName(),nodeId,uniqueID,actualtVal);
+
+				if(!res.IsSuccessful()) {
+					LOG_ERROR() << res.GetErrorMessage();
+				}*/
+
+			}
+
+
+
+		}
+	}
+
+
+	result1->release();
+
+}
+
+/**
+/Updates the Library with the information extracted from the node XDC.
+/param nodeid Specifies the ID of the Node.
+/param cnXDC Specifies the XDC path of Node.
+*/
+void LibraryUtils::UpdateNodeIdCollection(uint8_t nodeid, std::string cnXDC)
```

Comment:
new	app\src\LibraryUtils.cpp	324	performance	passedByValue	false	Function parameter 'cnXDC' should be passed by reference.